src/Tabs/Tab.tsx

```diff
@@ -4,11 +4,20 @@ import { StyledTab, StyledTabWrapper } from './Tab.styles';
 
 export interface TabProps {
   tabId: string;
+  beforeOnChange?: () => boolean;
 }
 
-export const Tab: React.FC<TabProps> = ({ tabId, children, ...rest }) => {
+export const Tab: React.FC<TabProps> = ({ tabId, beforeOnChange, children, ...rest }) => {
   const { setActiveTab, activeTab } = useTabsContext();
-  const handleClick = (): void => setActiveTab(tabId);
+  const handleClick = (): void => {
+    if (activeTab === tabId) return;
+
+    let shouldChange = true;
+    if (beforeOnChange) {
+      shouldChange = beforeOnChange();
+    }
+    shouldChange && setActiveTab(tabId);
```

Comment:
What about something like this (not the best code, just an idea)?

```const shouldChange = beforeOnChange?.();
    if (beforeOnChange && shouldChange) {
      setActiveTab(tabId);
    } else if (beforeOnChange && !shouldChange) {
      return false;
    } else {
      setActiveTab(tabId);
    }
```

And stipulate that `beforeOnChange` would return a boolean value or undefined. The thought is that we could then use the `beforeOnChange` either to block the tab change or not, basically allowing it to be passive or not. Happy to talk it through if you want?